pluginframework/resource_quality_monitor.go

```diff
@@ -0,0 +1,196 @@
+package pluginframework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/databricks/terraform-provider-databricks/service/catalog_tf"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+const qualityMonitorDefaultProvisionTimeout = 15 * time.Minute
+
+func WaitForMonitor(w *databricks.WorkspaceClient, ctx context.Context, monitorName string) error {
+	return retry.RetryContext(ctx, qualityMonitorDefaultProvisionTimeout, func() *retry.RetryError {
+		endpoint, err := w.QualityMonitors.GetByTableName(ctx, monitorName)
+		if err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		switch endpoint.Status {
+		case catalog.MonitorInfoStatusMonitorStatusActive:
+			return nil
+		case catalog.MonitorInfoStatusMonitorStatusError, catalog.MonitorInfoStatusMonitorStatusFailed:
+			return retry.NonRetryableError(fmt.Errorf("monitor status retrund %s for monitor: %s", endpoint.Status, monitorName))
+		}
+		return retry.RetryableError(fmt.Errorf("monitor %s is still pending", monitorName))
+	})
+}
+
+var _ resource.Resource = &QualityMonitorResource{}
+
+func ResourceQualityMonitor() func() resource.Resource {
+	return func() resource.Resource {
+		return &QualityMonitorResource{}
+	}
+}
+
+type QualityMonitorResource struct {
+	Client *common.DatabricksClient
+}
+
+func (r *QualityMonitorResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "databricks_lakehouse_monitor_pluginframework"
+}
+
+func (r *QualityMonitorResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Lakehouse Monitor. MonitorInfo struct is used to create the schema",
+		Attributes: common.PluginFrameworkResourceStructToSchemaMap(catalog_tf.MonitorInfo{}, func(c common.CustomizableSchemaPluginFramework) common.CustomizableSchemaPluginFramework {
+			c.AddNewField("skip_builtin_dashboard", schema.BoolAttribute{Optional: true, Required: false})
+			c.AddNewField("warehouse_id", schema.StringAttribute{Optional: true, Required: false})
+			c.SetRequired("assets_dir")
+			c.SetRequired("output_schema_name")
+			c.SetRequired("table_name")
+			// TODO: Uncomment this once SetReadOnly is supported in the plugin framework
```

Comment:
I don't know how to actually achieve this. As I saw in the provider schema types, all of the `isReadOnly` functions are hard coded to be returning false